### PR TITLE
READY: Generic cache

### PR DESCRIPTION
**Pull request by bschrock**
Anyone have input on the changes made and the direction this went?

### CODE REVIEW

File: lib/aws-cache.rb [hunk 1]
```diff
@@ -198,9 +156,22 @@ def has_key?(hash_or_struct, key)
     return false
   end
 
+  def cache_get_2(key, ttl)
```
Comment (reprehensible) [inline lib/aws-cache.rb:159]: is this functionally distinct from cache_get(), or just rewritten to be shorter/clearer?

**Comment by reprehensible** [comment]
Could you just write a couple of sentences about what the changes are trying to accomplish? The diff is a little hard to follow.

**Comment by bschrock** [comment]
The diff will be useless, I pretty much rewrote it the whole thing. The main purpose of this is to wrap AWS API calls and cache them in redis. I wrote this so that is keeps the responses from AWS pretty much intact. The API calls first get satisfied in redis and failing that will make API calls to get and store the data.

**Comment by reprehensible** [comment]
:+1: it looks good to me, assuming that it works :)
